### PR TITLE
feat: add clearning invitation users

### PR DESCRIPTION
**Pull request by fedyafedko**

### CODE REVIEW

File: StudyHub/StudyHub.BLL/Services/HangfireService.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    public bool DeleteUsers(List<InvitedUser> invitedUsers)
+    {
+        var invitedUserIds = invitedUsers.Select(invitedUser => invitedUser.Id).ToList();
+
+        var remainingUsers = _invitedUserRepository
+            .Where(u => invitedUserIds.Contains(u.Id))
+            .ToList();
+
+        return _invitedUserRepository.DeleteMany(remainingUsers);
```
Comment (Podgore) [inline StudyHub/StudyHub.BLL/Services/HangfireService.cs:24]: Are we needed there async, maybe?

**Review by Podgore — CHANGES_REQUESTED** [review]

**Review by Podgore — DISMISSED** [review]

File: StudyHub/StudyHub.Hangfire/HangfireConfig.cs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+﻿namespace StudyHub.Hangfire;
+public class HangfireConfig
+{
+    public string ClearingUserInvitationCron { get; set; }
```
Comment (Podgore) [inline StudyHub/StudyHub.Hangfire/HangfireConfig.cs:4]: some warnings here

File: StudyHub/StudyHub.Hangfire/Abstractions/IHangfireService.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿namespace StudyHub.Hangfire.Abstractions;
```
Comment (Podgore) [inline StudyHub/StudyHub.Hangfire/Abstractions/IHangfireService.cs:1]: empty line)

File: StudyHub/StudyHub.Hangfire/Abstractions/IJobArgs.cs
Comment (Podgore) [inline StudyHub/StudyHub.Hangfire/Abstractions/IJobArgs.cs:1]: are we need that empty interface?

File: StudyHub/StudyHub.Hangfire/Extensions/HangfireExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using StudyHub.Hangfire.Abstractions;
+using StudyHub.Hangfire.Jobs;
+
+namespace StudyHub.Hangfire.Extensions;
```
Comment (Podgore) [inline StudyHub/StudyHub.Hangfire/Extensions/HangfireExtensions.cs:7]: empty line

File: StudyHub/StudyHub.BLL/Services/UserInvitationService.cs [hunk 5]
```diff
@@ -21,28 +22,36 @@ public class UserInvitationService : IUserInvitationService
...
         _userManager = userManager;
-        _invitedUserRepository = invitedUserRepository;
-        _mapper = mapper;
         _roleManager = roleManager;
         _messageSettings = messageSettings.Value;
         _encryptService = encryptService;
+        _mapper = mapper;
+    }
+
+    public async Task ClearExpiredInvitations()
```
Comment (Podgore) [inline StudyHub/StudyHub.BLL/Services/UserInvitationService.cs:49]: maybe ClearExpiredInvintationsAsync?

**Review by Podgore — CHANGES_REQUESTED** [review]

**Review by Podgore — DISMISSED** [review]

File: StudyHub/StudyHub/Program.cs [hunk 6]
```diff
@@ -37,6 +44,10 @@
 builder.Services.AddDbContext<ApplicationDbContext>(opt =>
     opt.UseSqlServer(builder.Configuration.GetConnectionString("DefaultConnection")));
 
+// Hangfire
+builder.Services.AddHangfire(builder.Configuration);
+builder.Services.AddScoped<IHangfireService, HangfireService>();
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Program.cs:49]: move this up to AddHangfire extensions

File: StudyHub/StudyHub.BLL/Services/UserInvitationService.cs [hunk 7]
```diff
@@ -21,28 +21,36 @@ public class UserInvitationService : IUserInvitationService
...
         _roleManager = roleManager;
         _messageSettings = messageSettings.Value;
         _encryptService = encryptService;
+        _mapper = mapper;
+    }
+
+    public async Task ClearExpiredInvitationsAsync()
+    {
+        var expired = _invitedUserRepository
+            .Where(user => user.CreatedAt.AddDays(7) <= DateTime.Today);
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/UserInvitationService.cs:51]: move count of days (or lifetime in TimeSpan) to config

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by aquaprogit — APPROVED** [review]

**Review by Podgore — APPROVED** [review]

**Pull request merged**